### ISSUE TITLE
VOC doesn't support NetBSD

### ISSUE THREAD

**Issue by pascalecu** [vishapoberon/compiler#107]
Hi! I have been trying to compile voc, as oo2c has failed me massively. After cloning, I did `doas make full install`, but I was greeted with

```
** Unrecognised utsname.sysname 'NetBSD' returned by uname().
** Please add a test for this OS in src/buildtools/configure.c
** in function determineOS() near line 118.

Error: Unrecognised OS architecture name (sysname) returned by uname.
```

I will make a corresponding PR for this. It has been a simple fix (a single line) and I assume this was an oversight on your part. The instructions are wrong though, as not all BSDs have `pkg` (notably NetBSD) and there's nobody maintaining a voc port over on pkgsrc (I can be the maintainer).

**Comment by norayr** [comment]
hello, thank you for the pr!
did you test it?
i believe you also have to use gmake instead of make.

**Comment by pascalecu** [comment]
> hello, thank you for the pr! did you test it? i believe you also have to use gmake instead of make.

Yes, it compiles on NetBSD 10.1 at least. It didn't complain about BSD make and I don't see you using any GNU-isms, so I don't see why you'd need gmake. The problem was merely with the OS detection. I presume that's where I would have to go to add support for other Unix operating systems, but maybe I'm wrong and more changes are needed.

**Comment by pascalecu** [comment]
Proof:

```sh
$ pwd
/home/athena/voc
$ grep "netbsd" src/tools/make/configure.c
    else if (strncasecmp(sys.sysname, "netbsd", 5) == 0) {os = "netbsd"; bsd = 1;}
$ `which make` -f /dev/null -e -B -n
make: no target to make.

make: stopped in /home/athena/voc
$ `which gmake` -f /dev/null -e -B -n
gmake: *** No targets.  Stop.
$ doas make clean full install >& /dev/null; echo $?
Password:
0
$ set path = ( $path $PWD/install/bin )
$ voc

Oberon-2 compiler v2.1.0 [2025/03/12] for gcc LP64 on netbsd.
Based on Ofront by J. Templ and Software Templ OEG.
Further development by Norayr Chilingarian, David Brown and others.
Loaded from /home/athena/voc/install/bin

Usage:

  voc options {files {options}}.

Options:

  Run time safety
    -p   Initialise pointers to NIL. On by default.
    -a   Halt on assertion failures. On by default.
    -r   Halt on range check failures.
    -t   Halt on type guard failure. On by default.
    -x   Halt on index out of range. On by default.

  Symbol file management
    -e   Allow extension of old symbol file.
    -s   Allow generation of new symbol file.
    -F   Force generation of new symbol file.

  C compiler and linker control
    -m   This module is main. Link dynamically.
    -M   This module is main. Link statically.
    -S   Don't call C compiler
    -c   Don't link.

  Miscellaneous
    -f   Disable VT100 control characters in status output.
    -V   Display compiler debugging messages.

  Size model for elementary types (default O2)
    -O2   Original Oberon / Oberon-2:  8 bit SHORTINT, 16 bit INTEGER, 32 bit LONGINT and SET.
    -OC   Component Pascal:           16 bit SHORTINT, 32 bit INTEGER and SET, 64 bit LONGINT.
    -OV   Alternate large model:       8 bit SHORTINT, 32 bit INTEGER, 64 bit LONGINT and SET.

  Target machine address size and alignment (default is that of the running compiler binary)
    -A44   32 bit addresses, 32 bit alignment (e.g. Unix/linux 32 bit on x86).
    -A48   32 bit addresses, 64 bit alignment (e.g. Windows 32 bit on x86, linux 32 bit on arm).
    -A88   64 bit addresses, 64 bit alignment (e.g. 64 bit platforms).

All options are off by default, except where noted above.
Initial options specify defaults for all files.
Options following a filename are specific to that file.
Repeating an option toggles its value.
```
